package/run.sh

```diff
@@ -20,7 +20,27 @@ else
     FORCE_RESTART=false
 fi
 
-env "INSTALL_K3S_FORCE_RESTART=${FORCE_RESTART}" "INSTALL_K3S_SKIP_DOWNLOAD=true" "INSTALL_K3S_SKIP_SELINUX_RPM=true" "INSTALL_K3S_SELINUX_WARN=true" installer.sh $@
+
+# work around for https://github.com/k3s-io/k3s/issues/2306
+# depends on journald being available on first boot, so this may not work on all providers (e.g. Digital Ocean)
+DELAY=5
+while ! env "INSTALL_K3S_FORCE_RESTART=${FORCE_RESTART}" "INSTALL_K3S_SKIP_DOWNLOAD=true" "INSTALL_K3S_SKIP_SELINUX_RPM=true" "INSTALL_K3S_SELINUX_WARN=true" installer.sh $@
+do
+    # Only use the logs for the latest restart of k3s, otherwise errors encountered after an ETCD join error will not be reported properly.
+    K3S_LOGS=$(journalctl -u k3s.service)
+    LAST_LOGS="$K3S_LOGS"
+    LAST_START_LINE=$(echo "$K3S_LOGS" | grep "Starting k3s v1." -n  | cut -d: -f1 | tail -1)
```

Comment:
Instead of re-running the installer fully, I think it might be more valuable to directly manipulate the `systemd` unit for K3s. Re-running the installer is a bit brute force IMO, but if this is going to be too much work, we can just merge what you have already.

Reply:
I spent a good amount of time looking into how to do this but couldn't come up with a solid solution. If you have pointers on how it can be done properly I'm all ears, but i've kept it as is for now.